### PR TITLE
fix: secret function

### PR DESCRIPTION
**Pull request by kobenguyent**
## Motivation/Description of the PR
fix: secret function

````
  type redactedMyObj = { password: CodeceptJS.Secret };
  const myObj: any = secret({ password: 'pass123', pwd: 'hi' }, 'password') as redactedMyObj;
  console.log(myObj.password);

>>> Secret { _secret: 'pass123' }

  type redactedMyObj = { password: CodeceptJS.Secret };
  const myObj: any = secret({ password: 'pass123', pwd: 'hi' }, 'password') as redactedMyObj;
  console.log(myObj);
>>>> { password: Secret { _secret: 'pass123' }, pwd: 'hi' }



await I.sendPostRequest('/auth', secret({ name: 'jon', password: '123456' }, 'password'));
// console log
  Verify a successful call
    I send post request "/auth", {"name":"jon","password":"*****"}
    › [Request] {"baseURL":"https://reqres.in/auth","method":"POST","data":{"name":"jon","password":"*****"},"headers":{}}


````
## Type of change
- [ ] :bug: Bug fix

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [ ] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Review by roycollings — COMMENTED** [review]
@kobenguyent I think you can solve all of this by just changing the return value in the `get` handler in secret.js to `return fieldsToHide.includes(prop) ? new Secret(obj[prop]) : obj[prop];`:

```javascript
function secretObject(obj, fieldsToHide = []) {
  const handler = {
    get(obj, prop) {
      if (prop === 'toString') {
        return function () {
          const maskedObject = deepClone(obj);
          fieldsToHide.forEach(f => maskedObject[f] = maskStr);
          return JSON.stringify(maskedObject);
        };
      }

      // Just this change ...
      return fieldsToHide.includes(prop) ? new Secret(obj[prop]) : obj[prop];
    },
  };

  return new Proxy(obj, handler);
}
```

This makes the extracted item a secret - in my local tests this seems to work well (my tests can run login scripts etc.. with this approach):

_simple helper fn to demo:_
```typescript
  useMyFunc(secret_string, secret_object, secret_item_from_object) {
    console.log(">>>> types", {
      secret_string: typeof secret_string,
      secret_object: typeof secret_object,
      secret_item_from_object: typeof secret_item_from_object,
    });

    console.log(">>>> values", {
      secret_string,
      secret_item_from_object: secret_item_from_object,
    });
  }
```

_Send the secrets to see what's logged..._
```typescript
    const secret_string = secret("password");
    const secret_object = secret({ a: "pass123", b: "bob" }, "a");

    I.useMyFunc(secret_string, secret_object, secret_object.a);
```

_OUTPUT_
```bash
    I use my func *****, {"a":"****","b":"bob"}, *****
>>>> types {
  secret_string: 'object',
  secret_object: 'object',
  secret_item_from_object: 'object'
}
>>>> values {
  secret_string: Secret { _secret: 'password' },
  secret_item_from_object: Secret { _secret: 'pass123' }
}
```

**Comment by kobenguyent** [comment]
> @kobenguyent I think you can solve all of this by just changing the return value in the `get` handler in secret.js to `return fieldsToHide.includes(prop) ? new Secret(obj[prop]) : obj[prop];`:
> 
> ```js
> function secretObject(obj, fieldsToHide = []) {
>   const handler = {
>     get(obj, prop) {
>       if (prop === 'toString') {
>         return function () {
>           const maskedObject = deepClone(obj);
>           fieldsToHide.forEach(f => maskedObject[f] = maskStr);
>           return JSON.stringify(maskedObject);
>         };
>       }
> 
>       // Just this change ...
>       return fieldsToHide.includes(prop) ? new Secret(obj[prop]) : obj[prop];
>     },
>   };
> 
>   return new Proxy(obj, handler);
> }
> ```
> 
> This makes the extracted item a secret - in my local tests this seems to work well (my tests can run login scripts etc.. with this approach):
> 
> _simple helper fn to demo:_
> 
> ```ts
>   useMyFunc(secret_string, secret_object, secret_item_from_object) {
>     console.log(">>>> types", {
>       secret_string: typeof secret_string,
>       secret_object: typeof secret_object,
>       secret_item_from_object: typeof secret_item_from_object,
>     });
> 
>     console.log(">>>> values", {
>       secret_string,
>       secret_item_from_object: secret_item_from_object,
>     });
>   }
> ```
> 
> _Send the secrets to see what's logged..._
> 
> ```ts
>     const secret_string = secret("password");
>     const secret_object = secret({ a: "pass123", b: "bob" }, "a");
> 
>     I.useMyFunc(secret_string, secret_object, secret_object.a);
> ```
> 
> _OUTPUT_
> 
> ```shell
>     I use my func *****, {"a":"****","b":"bob"}, *****
> >>>> types {
>   secret_string: 'object',
>   secret_object: 'object',
>   secret_item_from_object: 'object'
> }
> >>>> values {
>   secret_string: Secret { _secret: 'password' },
>   secret_item_from_object: Secret { _secret: 'pass123' }
> }
> ```

Thanks @roycollings such an elegant fix than mine.

File: lib/helper/REST.js [hunk 1]
```diff
@@ -137,6 +137,15 @@ class REST extends Helper {
       request.auth = this.headers.auth;
     }
 
+    if (typeof request.data === 'object') {
+      let returnedValue = {};
```
Comment (roycollings) [inline lib/helper/REST.js]: Can be `const` (you're not mutating, just expanding :)

File: lib/step.js [hunk 2]
```diff
@@ -172,7 +172,12 @@ class Step {
       } else if (arg.toString && arg.toString() !== '[object Object]') {
         return arg.toString();
       } else if (typeof arg === 'object') {
-        return JSON.stringify(arg);
+        let returnedArg = {};
```
Comment (roycollings) [inline lib/step.js]: Can be `const`

**Review by roycollings — APPROVED** [review]
Couple of non-blockers :)

**Review by roycollings — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

**Pull request merged**